frontend/src/pages/UserPage.jsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from 'react'
+import userService from '../services/users'
+import UserInfo from '../components/UserInfo' 
+
+const UserPage = ({setDefaultRole}) => {
+  const [user, setUser] = useState({})
+
+  useEffect(() => {
+    userService.get().then(user => setUser(user))
+  }, [])
+
+  const handleRoleSwitch = () => {
+    const newRole = switchRole();
+    userService.setRole(newRole).then(_ => {
+      setUser({...user, role: newRole})
+      localStorage.setItem('role', newRole);
+      setDefaultRole(newRole)
```

Comment:
The same data is essentially stored in three different places. It works, but I am sure this can be done more efficiently at some other time.

Reply:
I agree, thanks for pointing that out!